server/src/test/java/org/opentosca/toscana/plugins/kubernetes/docker/mapper/MapperTest.java

```diff
@@ -0,0 +1,119 @@
+package org.opentosca.toscana.plugins.kubernetes.docker.mapper;
+
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.opentosca.toscana.core.BaseJUnitTest;
+import org.opentosca.toscana.core.testutils.ReflectionUtils;
+import org.opentosca.toscana.plugins.kubernetes.docker.OSCapability;
+import org.opentosca.toscana.plugins.kubernetes.docker.mapper.util.DataContainer;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.io.IOUtils;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.junit.Assert.assertEquals;
+import static org.opentosca.toscana.plugins.kubernetes.docker.mapper.DockerBaseImages.ALPINE;
+import static org.opentosca.toscana.plugins.kubernetes.docker.mapper.DockerBaseImages.DEBIAN;
+import static org.opentosca.toscana.plugins.kubernetes.docker.mapper.DockerBaseImages.UBUNTU;
+import static org.opentosca.toscana.plugins.kubernetes.docker.mapper.MapperConstants.DEFAULT_IMAGE_PATH;
+
+@RunWith(Parameterized.class)
+public class MapperTest extends BaseJUnitTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(MapperTest.class);
+    private static BaseImageMapper baseImageMapper;
+
+    private String name;
+    private OSCapability capability;
+    private String expectedImage;
+
+    public MapperTest(String name, OSCapability capability, String expectedImage) {
+        this.name = name;
+        this.capability = capability;
+        this.expectedImage = expectedImage;
+    }
+
+    @Test
+    public void test() throws Exception {
+        logger.info("Executing test {}", name);
+        String image = baseImageMapper.mapToBaseImage(capability);
+        logger.info("Mapped capability {} to {}. Expected is {}", capability.toString(), image, expectedImage);
+        assertEquals(expectedImage, image);
+    }
+
+    @BeforeClass
+    public static void initBaseImageMapper() throws Exception {
+        logger.info("Initializing Base image Mapper");
+
+        baseImageMapper = init();
+    }
+
+    public static BaseImageMapper init() throws Exception {
+        ObjectMapper mapper = new ObjectMapper();
+        InputStream in = BaseImageMapper.class.getClassLoader()
+            .getResourceAsStream("kubernetes/base-image-mapper/image-map.json");
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        IOUtils.copy(in, out);
+        BaseImageMapper baseImageMapper = new BaseImageMapper(new DockerBaseImages[] {
+            ALPINE,
+            DEBIAN,
+            UBUNTU
+        });
+        DataContainer data = mapper.readValue(new String(out.toByteArray()), DataContainer.class);
+
+        ReflectionUtils.setPrivateField(baseImageMapper, "imageMap", data.data);
```

Comment:
Testing private fields is an anti pattern. Tests should only test the external behaviour. If internal implementation changes, tests break. I suggest you get rid of the Reflection Util class and change imageMap visibility to default (package visiblity).

Reply:
This is not testing, this is used to bypass the Download mechanism.